libraries/tlv-account-resolution/src/state.rs

```diff
@@ -112,9 +121,36 @@ fn account_meta_from_info(account_info: &AccountInfo) -> AccountMeta {
 /// ExtraAccountMetaList::init::<MyInstruction>(&mut buffer, &extra_metas).unwrap();
 ///
 /// // Off-chain, you can add the additional accounts directly from the account data
+/// // You need to provide the resolver a way to fetch account data off-chain
+/// struct MyClient {
+///     client: RpcClient,
+/// }
+/// impl MyClient {
+///     pub fn new() -> Self {
+///         Self {
+///             client: RpcClient::new_mock("succeeds".to_string()),
+///         }
+///     }
+///     pub async fn get_account_data(&self, address: Pubkey) -> AccountDataResult {
+///         self.client.get_account(&address)
+///             .await
+///             .map(|acct| Some(acct.data))
+///             .map_err(|e| Box::new(e) as AccountFetchError)
+///     }
+/// }
+///
+/// let client = MyClient::new();
 /// let program_id = Pubkey::new_unique();
 /// let mut instruction = Instruction::new_with_bytes(program_id, &[0, 1, 2], vec![]);
-/// ExtraAccountMetaList::add_to_instruction::<MyInstruction>(&mut instruction, &buffer).unwrap();
+/// futures::executor::block_on(async {
+///     // Now use the resolver to add the additional accounts off-chain
+///     ExtraAccountMetaList::add_to_instruction::<MyInstruction, _, _>(
+///         &mut instruction,
+///         |address: Pubkey| client.get_account_data(address),
+///         &buffer,
+///     )
+///     .await;
+/// });
```

Comment:
nit: so that people reading the docs don't see the `futures::executor` bit, can we change this to
```suggestion
/// # futures::executor::block_on(async {
/// // Now use the resolver to add the additional accounts off-chain
/// ExtraAccountMetaList::add_to_instruction::<MyInstruction, _, _>(
///     &mut instruction,
///     |address: Pubkey| client.get_account_data(address),
///     &buffer,
/// )
/// .await;
/// # });
```